parlai/mturk/tasks/acute_eval/fast_eval.py

```diff
@@ -0,0 +1,626 @@
+            # Write the onboarding convo
+            pairs = onboarding_pairs + pairs
+            for pair in pairs:
+                f.write(json.dumps(pair) + "\n")
+
+    def _load_pairings_file(self):
+        """
+        Build the pairings file for ACUTE-Eval.
+
+        If a pairings file already exists, we ask the user whether they would like to
+        overwrite the pairings file.
+        """
+        pairings_filepath = self._get_vs_path('pairings_files')
+        # Rebuild pairings file if necessary
+        if not os.path.exists(pairings_filepath):
+            self._build_pairings_file()
+        else:
+            modify_time = os.path.getmtime(pairings_filepath)
+            self._print_progress(
+                f'Pairings already exist {pairings_filepath}. Last modified {time.ctime(modify_time)}'
+            )
+            answer = ''
+            while answer.lower().strip() != 'y' and answer.lower().strip() != 'o':
+                answer = input('Enter y to use, o to overwrite:')
+                if answer.lower().strip() == 'o':
+                    self._build_pairings_file()
+
+        # Load the file
+        self._print_progress(f'loading pairings file from {pairings_filepath}')
+        self.pairings_filepath = pairings_filepath
+
+    def _run_selfchat(self, config_id: str):
+        """
+        Run self-chat for model.
+
+        :param config_id:
+            id in config
+        """
+        self._print_progress(f'Running self-chat for {config_id}')
+        config = self._get_selfchat_config(config_id)
+
+        with capture_output():
+            parser = self_chat_setup_args()
+            parser.set_params(**config)
+            opt = parser.parse_args(args=[], print_args=False)
+        self_chat(opt)
+
+    def _convert_task_to_conversations(self, config_id: str):
+        """
+        Convert task data to conversations format.
+
+        :param config_id:
+            id in config
+        """
+        self._print_progress(
+            f'Converting task data to conversations format for {config_id}'
+        )
+        config = self._get_task_conversion_config(config_id)
+
+        with capture_output():
+            parser = convert_task_setup_args()
+            parser.set_params(**config)
+            opt = parser.parse_args(args=[], print_args=False)
+        convert_task_data(opt)
+
+    ##################
+    # Main Functions #
+    ##################
+    def compile_chat_logs(self):
+        """
+        Compile Chat Logs.
+
+        Logs are generated depending on what is specified in the config for the model:
+        1. If a `model` is provided, run selfchat for model
+        2. If a `log_path` is provided, simply load the log path
+        3. If a `task` is provided, convert the task to ACUTE format and load that.
+        """
+        for model in self.config_ids:
+            try:
+                torch.cuda.empty_cache()
+            except Exception:
```

Comment:
oo, what does this do?

Reply:
deletes the models from GPU, i found when I was first writing this that the models weren't being cleaned up correctly when self-chat was over, that might be fixed now but i'll leave it in for now